airflow/models/xcom.py

```diff
@@ -111,7 +111,7 @@ class BaseXCom(TaskInstanceDependencies, LoggingMixin):
         lazy="joined",
         passive_deletes="all",
     )
-    execution_date = association_proxy("dag_run", "execution_date")
```

Comment:
Same. Not sure if there are more (probably).

Reply:
Keeping it for now as discussed.